### ISSUE TITLE
Changing the texture of the normal/practice mode bar anywhere prevents it from cropping

### ISSUE THREAD

**Issue by JordanTRS** [Alphalaneous/HappyTextures#17]
Easiest way to explain this is to show a comparison. This is what it normally looks like:
![image](https://github.com/user-attachments/assets/7fc4a836-000f-4c36-8c0d-67808e66e170)
And this is what it looks like when I try to replace the sprite using HT:
![image](https://github.com/user-attachments/assets/daf94f6d-c594-41e2-a955-b30a8d867af4)

**Comment by Alphalaneous** [comment]
This isn’t exactly a bug but rather a quirk of how those sprites work in the first place. They work by having a full width sprite and then cropping it. Currently replacing a sprite doesn’t inherit the cropped rect of the sprite and it can be hard to make it exactly the same. This is something that should be done directly by replacing the full texture rather than changing the sprite in the json

**Comment by JordanTRS** [comment]
so there's no way to have separate sprites for each size of the progress bars? that's disappointing :/

**Comment by Alphalaneous** [comment]
As with the scroll thing, I can add a util for this in the future, as this would be cool for stylized packs

**Comment by JordanTRS** [comment]
Yeah, it would be especially helpful for packs that use pixel art since the progress bars in-game aren’t all the same size. There’s a lot that could be added but with time this could be one of the best GD mods out there
